src/api-handler.ts

```diff
@@ -127,34 +121,37 @@ export class ApiHandler implements MagicalClass {
      *
      * @param {*} args      Arguments
      * @returns {Promise}   Resolvable API provider promise
-     * @memberof ApiHandler
      */
     public async handleRequest(...args: any): Promise<IRequestResponse> {
         const prop = args[0];
-        const api = this.apiEndpoints[prop];
+        const endpointSettings = this.apiEndpoints[prop];
 
         const queryParams = args[1] || {};
         const uriParams = args[2] || {};
         const requestConfig = args[3] || {};
 
-        const uri = api.url.replace(/:[a-z]+/ig, (str: string) => (uriParams[str.substr(1)] ? uriParams[str.substr(1)] : str));
-
-        const requestData = {
-            ...queryParams,
-        };
+        const uri = endpointSettings.url.replace(/:[a-z]+/ig, (str: string) => (uriParams[str.substr(1)] ? uriParams[str.substr(1)] : str));
 
         let responseData = null;
 
-        responseData = await this.httpRequestHandler[api.method](uri, requestData, requestConfig);
+        const additionalRequestSettings = { ...endpointSettings };
+
+        delete additionalRequestSettings.url;
+        delete additionalRequestSettings.method;
```

Comment:
Now with the delete part, as much it is fine, you could ask well select properties you really need from `endpointSettings ` and then leave the others as `...others` 

Proper example 
```
const *objRef = {
   propertKey1, 
   propertkey2
}

const { propertKey1, propertkey2, ...others } = &objRef

// use your extracted properties
cont >> propertKey1
cont >> propertkey2
```

Reply:
Yes that would be more explicit. I'll make some tests and push for improvements in next release. PRs are welcomed 🙌

Reply:
As soon I've time, I'll help u into further integration of this lib.